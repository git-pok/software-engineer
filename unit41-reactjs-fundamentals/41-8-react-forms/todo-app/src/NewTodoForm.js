import { useState } from 'react';
import { v4 as uuid } from 'uuid';
import './NewTodoForm.css';

/**
 * NewTodoForm - this component should render a form
 * with one text input for the task to be created. When
 * this form is submitted, a new Todo component
 * should be created.
 */


const NewTodoForm = ({ addTodo }) => {

  const INITIAL_STATE = {
    todo: ""
  };

  const [ formData, setFormData ] = useState(INITIAL_STATE);

  const handleChange = e => {
    const evtTarget = e.target; 
    const { name, value } = evtTarget;

    setFormData(formData => ({
      ...formData,
      [name]: value
    }));
  }

  const handleSubmit = e => {
    e.preventDefault();
    addTodo({...formData, id: uuid()});
    setFormData(INITIAL_STATE);
  }

  return (
    <form onSubmit={handleSubmit} className="NewTodoForm-form">
      <label
        className="NewTodoForm-label"
        htmlFor="todo">Todo</label>
      <input
        className="NewTodoForm-input"
        id="todo"
        type="text"
        name="todo"
        placeholder="Create Todo!!!"
        value={formData.todo}
        onChange={handleChange} />

      <button className="NewTodoForm-submit">ADD TODO</button>
    </form>
  );
}

export default NewTodoForm;