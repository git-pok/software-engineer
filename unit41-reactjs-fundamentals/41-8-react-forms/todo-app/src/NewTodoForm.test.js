import { render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm.js'; 


test('<NewTodoForm /> renders without crashing', () => {
  render(<NewTodoForm />);
});


test('fragment matches snapshot', () => {
  const { asFragment } = render(<NewTodoForm />);
  expect(asFragment()).toMatchSnapshot();
});