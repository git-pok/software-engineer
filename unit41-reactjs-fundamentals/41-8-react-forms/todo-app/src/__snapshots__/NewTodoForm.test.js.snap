// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fragment matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTodoForm-form"
  >
    <label
      class="NewTodoForm-label"
      for="todo"
    >
      Todo
    </label>
    <input
      class="NewTodoForm-input"
      id="todo"
      name="todo"
      placeholder="Create Todo!!!"
      type="text"
      value=""
    />
    <button
      class="NewTodoForm-submit"
    >
      ADD TODO
    </button>
  </form>
</DocumentFragment>
`;
